### PR TITLE
refactor: update labels in benchmark terraform script

### PR DESCRIPTION
**Pull request by kruskall**
## Motivation/summary

Use the module/tags module for concistency across resources. 
Set created_date label.
Fix coalesce being unused.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## Related issues

Related https://github.com/elastic/apm-server/issues/9699

### CODE REVIEW

File: testing/benchmark/main.tf
Comment (kruskall) [inline testing/benchmark/main.tf:18]: `coalesce` wasn't working because environment and repo are never null/empty since they have a default value.

File: testing/benchmark/main.tf
Comment (kruskall) [inline testing/benchmark/main.tf:24]: repo is basically a duplicate of the team tag in the case of apm-server. Is it worth adding it ?

File: testing/benchmark/main.tf
Comment (kruskall) [inline testing/benchmark/main.tf:37]:
```terraform
module "tags" {
  source  = "../infra/terraform/modules/tags"
  project = "benchmarks"
}
```

I'm not sure about this. We're setting `project = benchmarks`, using the convention for `a team/shared owned resource` but benchmarks are actually always individually owned. Am I missing something ?

Comment (marclop) [reply testing/benchmark/main.tf:37]: That makes sense when the benchmark is launched locally, but if it's launched as a CI run, then that doesn't hold anymore since some resources may be left dangling, and we own it as a team since the CI scheduled run can't own that.

File: testing/infra/terraform/modules/benchmark_executor/instance.tf
Comment (marclop) [inline testing/infra/terraform/modules/benchmark_executor/instance.tf:48]: `Name` is a special tag that is used for resource naming, it should have the Title format.

```suggestion
    Name = "vpc-${var.user_name}-worker"
```

**Review by marclop — APPROVED** [review]

File: testing/benchmark/Makefile
Comment (marclop) [inline testing/benchmark/Makefile:57]: This will break `make terraform.tfvars` locally, since the current user won't be used for prefixes, I'd rather keep it as it was.

File: testing/benchmark/main.tf
Comment (kruskall) [reply testing/benchmark/main.tf:39]: I've pushed some changes to account for CI! Let me know what you think!

File: testing/benchmark/Makefile
Comment (kruskall) [reply testing/benchmark/Makefile]: Updated!

File: testing/infra/terraform/modules/benchmark_executor/variables.tf
Comment (marclop) [inline testing/infra/terraform/modules/benchmark_executor/variables.tf:39]:
```suggestion
  description = "Optional set of tags to use for all resources"
```

File: testing/infra/terraform/modules/benchmark_executor/instance.tf
Comment (marclop) [inline testing/infra/terraform/modules/benchmark_executor/instance.tf:55]: I think all AWS resources should have the tags set in `ec2_tags`. Given that, it would be easier to define the `provider "aws"` block within this module, so all created resources have the same tags consistently. This will already happen in `testing/benchmark` since the module is consumed by that folder and we define a provider block, however, for other users, we can enforce the tag standard within the module consistently.

**Review by marclop — COMMENTED** [review]
We should set `default_tags` in `testing/benchmark`, and the `benchmark_executor` module, so other importers of this module also have the tags set for them.

File: testing/infra/terraform/modules/benchmark_executor/instance.tf
Comment (kruskall) [reply testing/infra/terraform/modules/benchmark_executor/instance.tf]: I've added `ec2_tags` to all resources

File: testing/benchmark/main.tf
Comment (kruskall) [reply testing/benchmark/main.tf:43]: I tired to use `default_tags` and it's only applying to the `aws_key_pair`: https://github.com/elastic/apm-server/blob/main/testing/infra/terraform/modules/benchmark_executor/instance.tf#L42

It seems the other resources are using a different provider (https://registry.terraform.io/modules/terraform-aws-modules/vpc/aws/latest for example) so we need to set tags explicitly.

Let me know what you think!

File: testing/benchmark/main.tf
Comment (marclop) [inline testing/benchmark/main.tf:44]: Did you mean to remove the entire `provider` block? AFAIK we still need it so we can specify the worker region. Before we merge, can you make sure trigger the benchmarks and that they pass? It should also be a good way to see the exact tags that are applied.

Comment (kruskall) [reply testing/benchmark/main.tf]: Good catch! I've added back the block.

I've also confirmed that the terraform script are working and the benchmarks run :+1:

**Comment by marclop** [comment]
I triggered the benchmarks on the CI and they're currently failing, can you please take a look?

**Comment by kruskall** [comment]
> I triggered the benchmarks on the CI and they're currently failing, can you please take a look?

@marclop Thanks for that! It seems ci was using an old version of terraform. I've updated the version

**Comment by marclop** [comment]
Please don't merge the PR until the `apm-server-benckmarks/pr-merge` check is passing.

**Comment by kruskall** [comment]
> Please don't merge the PR until the apm-server-benckmarks/pr-merge check is passing. If we do, we'll break the daily benchmarks. https://github.com/orgs/elastic/teams/observablt-ci Could check why the updated Terraform version is not being used in the CI runner?

@marclop Thanks for mentioning this! I've pinged them on slack, see the related channel for the root cause. I think the issue is solved now :+1:

**Comment by kruskall** [comment]
`apm-server-benckmarks/pr-merge` is green!

Thank you @v1v for the super fast response! :bow:

**Pull request merged**
